pkg/virt-operator/resource/apply/core_test.go

```diff
@@ -395,24 +396,19 @@ var _ = Describe("Apply", func() {
 
 				Expect(hasImmutableFieldChanged(targetService, cachedService)).To(BeFalse())
 				ops, err := generateServicePatch(cachedService, targetService)
-				Expect(err).ToNot(HaveOccurred())
-
-				hasSubstring := func(ops []string, substring string) bool {
-					for _, op := range ops {
-						if strings.Contains(op, substring) {
-							return true
-						}
-					}
-					return false
+				if !expectLabelsAnnotationsPatch && !expectSpecPatch {
+					Expect(ops).To(BeEmpty())
+				} else {
 				}
+				Expect(err).ToNot(HaveOccurred())
 
 				if expectLabelsAnnotationsPatch {
-					Expect(hasSubstring(ops, "/metadata/labels")).To(BeTrue())
-					Expect(hasSubstring(ops, "/metadata/annotations")).To(BeTrue())
+					Expect(strings.Contains(string(ops), "/metadata/labels")).To(BeTrue())
+					Expect(strings.Contains(string(ops), "/metadata/annotations")).To(BeTrue())
```

Comment:
```suggestion
					Expect(string(ops)).To(ContainSubstring("/metadata/labels"))
					Expect(string(ops)).To(ContainSubstring("/metadata/annotations"))
```
same for the `expectSpecPatch` below